minimint-api/src/db/mod.rs

```diff
@@ -45,7 +45,41 @@ pub trait Database: Send + Sync {
 
     fn raw_find_by_prefix(&self, key_prefix: &[u8]) -> PrefixIter<'_>;
 
-    fn raw_apply_batch(&self, batch: DbBatch) -> Result<()>;
+    fn transaction<'a>(&'a self) -> Result<Box<dyn Transaction<'a>>>;
+}
+
+pub trait Transaction<'a>: 'a {
+    fn raw_insert_entry(&mut self, key: &[u8], value: Vec<u8>) -> Result<Option<Vec<u8>>>;
```

Comment:
What's the meaning of `raw_` prefix? Should it be `insert_raw_entry`?

Reply:
BTW. Lots of databases are using `upsert` for "insert or update" and I find this `insert_new` kind of odd, but I guess it comes from some underlying databases naming?